src/protocol/dp/distributions.rs

```diff
@@ -0,0 +1,118 @@
+use rand::{
+    distributions::{Distribution, Uniform},
+    Rng,
+};
+
+use std::{f64::consts::PI, fmt::Debug};
+
+#[cfg(all(test, unit_test))]
+pub fn close(a: f64, b: f64, precision: u8) -> bool {
+    (a - b).abs()
+        < (2.0_f64.powf((a.abs() + 1.0).log2().ceil()) / 10.0_f64.powi(i32::from(precision)))
+}
+
+/// Normal distribution based on [`Box-Muller`] transform.
+///
+/// [`Box-Muller`]: https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform
+#[derive(Debug)]
+pub struct BoxMuller {
+    pub mean: f64,
+    pub std: f64,
+}
+
+impl Distribution<f64> for BoxMuller {
+    fn sample<R>(&self, rng: &mut R) -> f64
+    where
+        R: ?Sized + Rng,
+    {
+        let ud = Uniform::new(0.0, 1.0);
+        let u = ud.sample(rng);
+        let v = ud.sample(rng);
+        let n = f64::sqrt(-2.0 * f64::ln(u)) * f64::cos(2.0 * PI * v);
+
+        // map sample to N(mean,variance)=sqrt(variance)*sample+mean
+        n * self.std + self.mean
+    }
+}
+
+/// Rounded Normal distribution based on [`Box-Muller`] transform.
+///
+/// [`Box-Muller`]: https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform
+#[derive(Debug)]
+pub struct RoundedBoxMuller {
+    pub mean: f64,
+    pub std: f64,
```

Comment:
This should hold an instance of `BoxMuller` rather than re-implementing the algorithm.